### PR TITLE
Snowflake usage crawler error [sc-7154]

### PR DESCRIPTION
**Pull request by alyiwang**
### 🤔 Why?

pydantic validation does not allow unexpected fields in a model. Previous python build-in dataclass simply ignores extra fields. 

The Snowflake access history has different object structures https://docs.snowflake.com/en/user-guide/access-history.html , especially when it's copying for a STAGE or a local file. 

### 🤓 What?

- Add potential fields of access history log in the model as optional with default None
- Check for only table/view `objectDomain` before further processing

### 🧪 Tested?

tested against snowflake instances.

### CODE REVIEW

File: metaphor/snowflake/usage/extractor.py [hunk 1]
```diff
@@ -31,10 +32,13 @@ class AccessedObjectColumn:
...
-    objectName: str
-    objectId: int
-    columns: List[AccessedObjectColumn]
+    objectDomain: str = ""
+    objectName: str = ""
+    objectId: int = 0
+    stageKind: Optional[str] = None
+    columns: List[AccessedObjectColumn] = field(default_factory=lambda: list())
+    location: Optional[str] = None
+    locations: Optional[List[str]] = None
```
Comment (elic-eon) [inline metaphor/snowflake/usage/extractor.py:41]: Did Pydantic provide the option to drop extra fields?

Comment (mars-lan) [reply metaphor/snowflake/usage/extractor.py:41]: +1. It might be more future-proof to ignore extra fields.

Comment (alyiwang) [reply metaphor/snowflake/usage/extractor.py]: Tried this initially last night following the pydantic config https://pydantic-docs.helpmanual.io/usage/model_config/#options, but for a long time I couldn't get it to work and finally gave up. The default `extra=ignore` is already what we want, but its utility function just doesn't take this config and still enforce no extra field. Even explicitly set this config doesn't help

Comment (elic-eon) [reply metaphor/snowflake/usage/extractor.py]: Great, maybe we should also have our configs inherited from the `BaseModel` to have better support.

**Review by elic-eon — APPROVED** [review]

**Pull request merged**